# Шаг 6 - Проверка размерности данных

Поскольку на предыдущем этапе данные были разделены на два уровня детализации (уровень "набор" и уровень "набор-национальность"), проверка размерностьи будет осуществлена для обоих итоговых файлов.

In [1]:
import pandas as pd

# 1. Загружаем итоговые консолидированные файлы
df_sets = pd.read_csv('df_sets.csv')
df_consolidated = pd.read_csv('df_consolidated.csv')

# 2. Проверяем размерность (количество строк и столбцов)
print("=== Размерность данных ===")
print(f"df_sets (уровень набора): {df_sets.shape[0]} строк, {df_sets.shape[1]} столбцов")
print(f"df_consolidated (уровень набора + национальность): {df_consolidated.shape[0]} строк, {df_consolidated.shape[1]} столбцов")

# 3. Быстрый визуальный контроль структуры
print("\n=== Первые 3 строки df_sets ===")
display(df_sets.head(3))

print("\n=== Первые 3 строки df_consolidated ===")
display(df_consolidated.head(3))

=== Размерность данных ===
df_sets (уровень набора): 2840 строк, 13 столбцов
df_consolidated (уровень набора + национальность): 3937 строк, 14 столбцов

=== Первые 3 строки df_sets ===


,ID,Header,Years_from,Years_to,Release_Year,Aggregate_Rating,Num_Figures,Manufacturer,Period_Duration,Mid_Year,Era,Decade,Years_Since_Release
0,1,Accurate British Infantry (7200),1768,1796,NaN,47.0,50.0,Accurate,28,1782,Новое время,NaN,NaN
1,2,Accurate American Militia (7201),1775,1780,NaN,48.0,50.0,Accurate,5,1777,Новое время,NaN,NaN
2,3,Accurate Union Infantry (7202),1861,1865,1991.0,48.0,50.0,Accurate,4,1863,Новое время,1990.0,35.0



=== Первые 3 строки df_consolidated ===


,ID,Header,Years_from,Years_to,Release_Year,Aggregate_Rating,Num_Figures,Manufacturer,Period_Duration,Mid_Year,Era,Decade,Years_Since_Release,Nationality
0,1,Accurate British Infantry (7200),1768,1796,NaN,47.0,50.0,Accurate,28,1782,Новое время,NaN,NaN,British
1,2,Accurate American Militia (7201),1775,1780,NaN,48.0,50.0,Accurate,5,1777,Новое время,NaN,NaN,USA
2,3,Accurate Union Infantry (7202),1861,1865,1991.0,48.0,50.0,Accurate,4,1863,Новое время,1990.0,35.0,USA


## Результат шага
Количество строк и столбцов проверено и соответствует ожидаемому масштабу данных после консолидации. Потерь данных на этапе объединения не произошло (количество уникальных ID в df_sets равно 2840).